flink-dist/src/main/flink-bin/bin/config.sh

```diff
@@ -334,6 +334,12 @@ if [ -z "${FLINK_ENV_JAVA_OPTS}" ]; then
 
     # Remove leading and ending double quotes (if present) of value
     FLINK_ENV_JAVA_OPTS="-XX:+IgnoreUnrecognizedVMOptions $( echo "${FLINK_ENV_JAVA_OPTS}" | sed -e 's/^"//'  -e 's/"$//' )"
+
+    JAVA_SPEC_VERSION=`"${JAVA_RUN}" -XshowSettings:properties 2>&1 | grep "java.specification.version" | cut -d "=" -f 2 |  tr -d '[:space:]'`
+    if [[ $(echo "$JAVA_SPEC_VERSION > 17" | bc )  == "1"  ]]; then
```

Comment:
bc isn't available in our CI image. any reason this cant use bash arithmetic?

Reply:
yes, I should have been checking that failure
 the issue was with jdk 8 for which it returns `1.8`
 
 finally i replaced `bc` with usage of substring from latest dot till end